build-system/tasks/helpers.js

```diff
@@ -374,96 +372,120 @@ function finishBundle(srcFilename, destDir, destFilename, options) {
+  const startTime = Date.now();
   const entryPoint = path.join(srcDir, srcFilename);
   const destFilename = options.toName || srcFilename;
-  const wrapper = options.wrapper || wrappers.none;
-  const devWrapper = wrapper.replace('<%= contents %>', '$1');
-
-  // TODO: @jonathantyng remove browserifyOptions #22757
-  const browserifyOptions = {
-    entries: entryPoint,
-    debug: true,
-    fast: true,
-    ...options.browserifyOptions,
-  };
-
-  let bundler = browserify(browserifyOptions).transform(babelify, {
-    caller: {name: 'unminified'},
-    global: true,
-  });
+  const destFile = path.join(destDir, destFilename);
 
   if (options.watch) {
-    const watchFunc = () => {
-      const bundleComplete = performBundle(/* continueOnError */ true);
+    const watchFunc = async () => {
+      const bundleDone = await performBundle(/* continueOnError */ true);
       if (options.onWatchBuild) {
-        options.onWatchBuild(bundleComplete);
+        options.onWatchBuild(bundleDone);
       }
     };
-    bundler = watchify(bundler);
-    bundler.on('update', debounce(watchFunc, watchDebounceDelay));
+    fileWatch(entryPoint).on('change', debounce(watchFunc, watchDebounceDelay));
   }
 
   /**
-   * @param {boolean} continueOnError
-   * @return {Promise}
+   * Babel plugin for esbuild
    */
-  function performBundle(continueOnError) {
-    let startTime;
-    return toPromise(
-      bundler
-        .bundle()
-        .once('readable', () => (startTime = Date.now()))
-        .on('error', (err) =>
-          handleBundleError(err, continueOnError, destFilename)
-        )
-        .pipe(source(srcFilename))
-        .pipe(buffer())
-        .pipe(sourcemaps.init({loadMaps: true}))
-        .pipe(
-          regexpSourcemaps(
-            /\$internalRuntimeVersion\$/g,
-            internalRuntimeVersion,
-            'runtime-version'
-          )
-        )
-        .pipe(regexpSourcemaps(/([^]+)/, devWrapper, 'wrapper'))
-        .pipe(rename(destFilename))
-        .pipe(sourcemaps.write('.'))
-        .pipe(gulp.dest(destDir))
-        .on('end', () =>
-          finishBundle(srcFilename, destDir, destFilename, options)
-        )
-    )
-      .then(() => {
-        let name = destFilename;
-        if (options.latestName) {
-          const latestMaxName = options.latestName.split('.js')[0] + '.max.js';
-          name = `${name} → ${latestMaxName}`;
-        }
-        endBuildStep('Compiled', name, startTime);
-      })
-      .then(() => {
-        const target = path.basename(destFilename, path.extname(destFilename));
-        if (UNMINIFIED_TARGETS.includes(target)) {
-          return applyAmpConfig(
-            `${destDir}/${destFilename}`,
-            /* localDev */ true,
-            /* fortesting */ options.fortesting
-          );
+  const esbuildBabelPlugin = {
+    name: 'babel',
+    async setup(build, {transform} = {}) {
+      const updateVersion = (contents) => {
```

Comment:
Why isn't this just done with a babel plugin?

Reply:
Because I was trying to minimize changes in this PR 😃 

I'm working on a follow up that tries to make `$internalRuntimeVersion$` a const variable instead of a literal string. If that's a bad idea due to how closure works, then making it a babel plugin SGTM.

Reply:
I said similarly in a different comment. addressing in a followup

Reply:
Turns out we can't make this a `const` because we'd then have to re-jig how tests run. Right now, we conveniently assume that the string is not replaced for tests. Changing this to a `const` yields this new failure:
```sh
  ReferenceError: INTERNAL_RUNTIME_VERSION is not defined
      at internalRuntimeVersion (/var/folders/nv/ycwrjg_j32gdz6s542xlxy0m0037zm/T/src/internal-version.js:27:3)
      at getRtvVersion (/var/folders/nv/ycwrjg_j32gdz6s542xlxy0m0037zm/T/src/mode.js:135:15)
      at getMode_ (/var/folders/nv/ycwrjg_j32gdz6s542xlxy0m0037zm/T/src/mode.js:85:18)
      at getMode (/var/folders/nv/ycwrjg_j32gdz6s542xlxy0m0037zm/T/src/mode.js:54:28)
      at new Log (/var/folders/nv/ycwrjg_j32gdz6s542xlxy0m0037zm/T/src/log.js:155:16)
      at dev (/var/folders/nv/ycwrjg_j32gdz6s542xlxy0m0037zm/T/src/log.js:834:22)
      at initLogConstructor (/var/folders/nv/ycwrjg_j32gdz6s542xlxy0m0037zm/T/src/log.js:757:3)
      at Object.require.702.../build/ampdoc.css (/var/folders/nv/ycwrjg_j32gdz6s542xlxy0m0037zm/T/src/runtime.js:51:1)
      at o (/var/folders/nv/ycwrjg_j32gdz6s542xlxy0m0037zm/T/node_modules/browser-pack/_prelude.js:1:1)
      at /var/folders/nv/ycwrjg_j32gdz6s542xlxy0m0037zm/T/node_modules/browser-pack/_prelude.js:1:1

```

Can I leave it to one of y'all to move this into a babel plugin?